.github/workflows/pr.yaml

```diff
@@ -50,75 +111,152 @@ jobs:
       run_script: "ci/build_docs.sh"
+      default_endpoint: "${{needs.telemetry-setup.outputs.endpoint}}"
+      traceparent: ${{ needs.telemetry-setup.outputs.traceparent }}
+      otel_resource_attributes: "${{needs.telemetry-setup.outputs.otel_resource_attributes}}"
   wheel-build-cpp:
-    needs: checks
+    needs:
+      - checks
+      - telemetry-setup
     secrets: inherit
-    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-24.12
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@add-telemetry
     with:
       matrix_filter: group_by([.ARCH, (.CUDA_VER|split(".")|map(tonumber)|.[0])]) | map(max_by(.PY_VER|split(".")|map(tonumber)))
       build_type: pull-request
       script: ci/build_wheel_cpp.sh
+      default_endpoint: "${{needs.telemetry-setup.outputs.endpoint}}"
+      traceparent: ${{ needs.telemetry-setup.outputs.traceparent }}
+      otel_resource_attributes: "${{needs.telemetry-setup.outputs.otel_resource_attributes}}"
   wheel-build-python:
-    needs: wheel-build-cpp
+    needs:
+      - wheel-build-cpp
+      - telemetry-setup
     secrets: inherit
-    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-24.12
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@add-telemetry
     with:
       build_type: pull-request
       script: ci/build_wheel_python.sh
+      default_endpoint: "${{needs.telemetry-setup.outputs.endpoint}}"
+      traceparent: ${{ needs.telemetry-setup.outputs.traceparent }}
+      otel_resource_attributes: "${{needs.telemetry-setup.outputs.otel_resource_attributes}}"
   wheel-tests:
-    needs: [wheel-build-python, changed-files]
+    needs: [wheel-build-python, changed-files, telemetry-setup]
     secrets: inherit
-    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.12
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@add-telemetry
     if: fromJSON(needs.changed-files.outputs.changed_file_groups).test_python
     with:
       build_type: pull-request
       script: ci/test_wheel.sh
+      default_endpoint: "${{needs.telemetry-setup.outputs.endpoint}}"
+      traceparent: ${{ needs.telemetry-setup.outputs.traceparent }}
+      otel_resource_attributes: "${{needs.telemetry-setup.outputs.otel_resource_attributes}}"
   devcontainer:
     secrets: inherit
-    uses: rapidsai/shared-workflows/.github/workflows/build-in-devcontainer.yaml@branch-24.12
+    uses: rapidsai/shared-workflows/.github/workflows/build-in-devcontainer.yaml@add-telemetry
+    needs:
+      - telemetry-setup
     with:
       arch: '["amd64"]'
       cuda: '["12.5"]'
+      default_endpoint: "${{needs.telemetry-setup.outputs.endpoint}}"
+      traceparent: ${{ needs.telemetry-setup.outputs.traceparent }}
       build_command: |
         sccache -z;
         build-all -DBUILD_BENCHMARKS=ON --verbose;
         sccache -s;
+  final-telemetry-update:
```

Comment:
will this job and the `telemetry-setup` eventually be moved to a shared workflow?

Reply:
probably a shared action, not a shared workflow, but yes. that might help improve the DRY stuff here.